### ISSUE TITLE
Reload data in recyclerview

### ISSUE THREAD

**Issue by krtec** [h6ah4i/android-advancedrecyclerview#188]
Hi, i have recyclerview with advanced recycler view for drag and drop system. When i create recycler with data that is OK, but if i reload a data by 

        mRecyclerView.swapAdapter(gridviewAdapterItems, false);
        gridviewAdapterItems = new GridViewAdapterItems(products, MyApplication.getAppContext());
        mRecyclerView.setAdapter(gridviewAdapterItems);
        mRecyclerView.invalidate();

so data is ok loaded but system drag and drop not works perfect https://vid.me/clFY (drag and drop works only before and next item)
Thanks Jakub

**Comment by h6ah4i** [comment]
Hi. If you want to swap data set by using `setAdapter()`/`swapAdapter()` , need to recreate a wrapped adapter again.

```java
// clean up
WrapperAdapterUtils.releaseAll(mWrappedAdapter);
mRecyclerViewDragDropManager.release();

GridViewAdapterItems gridviewAdapterItems = new GridViewAdapterItems(products, MyApplication.getAppContext());

// create wrapped adapter
mRecyclerViewDragDropManager = new RecyclerViewDragDropManager();
mWrappedAdapter = mRecyclerViewDragDropManager.createWrappedAdapter(gridviewAdapterItems);

mRecyclerView.swapAdapter(mAdapter, false);

// setup drag & drop feature
mRecyclerViewDragDropManager.attachRecyclerView(mRecyclerView);
```

Also, `setItems()` and `notifyDataSetChanged()` method approach can be used.

```java
gridviewAdapterItems.setItems( newDataSet );
gridviewAdapterItems.notifyDataSetChanged();
```

**Comment by krtec** [comment]
Yes, thanks you so much, when i used this code, i got error java.lang.IllegalStateException: adapter is not set properly but i add 

mRecyclerView.setAdapter(mWrappedAdapter);

and now is perfects, thanks you so much. Jakub :)

**Comment by wandering7man** [comment]
Hi,i have 2 questions.
Ii have recyclerview with expanded recycler view and:
1) how can i update my data in list? sample above don't help me, I got exception;(
2) how can i keep state of opened/closed headers after list was updated(maybe only childs, or some headers, e.t.c)?it possible?
Thank you.

**Comment by h6ah4i** [comment]
@wandering7man Hi.

Hi. I have recently got a same question via email and I created a demo code how to refresh items in expandable list.

Please clone, build and run the example app of the following branch. Reset, shuffle and clear buttons will be added to the "Expandable" demo.

Branch: https://github.com/h6ah4i/android-advancedrecyclerview/compare/demo_update_expandable
Commit: ["Add reset/shuffle/clear buttons to expandable demo"](https://github.com/h6ah4i/android-advancedrecyclerview/commit/040982724b0f126746ce2ae1190ff98c5fcab28f)

<img src="https://cloud.githubusercontent.com/assets/2552365/13481063/d51e0726-e125-11e5-90d4-141f0717ae6d.png" width="250" />

**Comment by wandering7man** [comment]
@h6ah4i, wow thank you, you saved my day :+1:
